### PR TITLE
FEATURE: SD-2929 expose function for demo workspace setting

### PR DESCRIPTION
**Pull request by thuytran-gooddata**
-  Add function addTheme, addColorPalette
-  Expose workspaceDataFilter, workspaceDataFilterSettings for workspace 
JIRA: FEATURE: SD-2929

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by thuytran-gooddata** [comment]
ok to test

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15681/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17206/ : SUCCESS in 15m 43s

File: libs/sdk-backend-spi/src/workspace/settings/index.ts [hunk 1]
```diff
@@ -29,4 +29,22 @@ export interface IWorkspaceSettingsService {
...
+     * @param colorPaletteId - ID of the color palette to apply to charts in workspace.
+     *
+     * @returns promise
+     */
+    setColorPalette(colorPaletteId: string): Promise<void>;
+
+    /**
+     * Sets theme for current workspace.
+     *
+     * @param themeId - ID of the theme to apply to workspace in organization.
```
Comment (anvuong119) [inline libs/sdk-backend-spi/src/workspace/settings/index.ts:45]: I think `in organization` might make people think that the function can be used to set a theme ID to all workspaces in an organization, which is not the case, is it?

→ Please update the comment as below instead:

```
@param themeId - ID of the theme to apply to the current workspace.
```

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 2]
```diff
@@ -1210,6 +1247,67 @@ export const buildTigerSpecificFunctions = (
...
+    setWorkspaceDataFilters: async (
+        workspaceId: string,
+        workspaceDataFilter: WorkspaceDataFilter,
+    ): Promise<WorkspaceDataFiltersResult> => {
+        return authApiCall(async (sdk) => {
+            const result = await sdk.entities.createEntityWorkspaceDataFilters({
+                workspaceId,
+                jsonApiWorkspaceDataFilterInDocument: workspaceDataFilter,
+            });
+            return result.data;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:1277]: Do we really need to return value for a `setXXX` function? 

There is also no return value for function [setWorkspaceDataFiltersLayout](https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts#L1206C5-L1206C34), is there?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 3]
```diff
@@ -1210,6 +1247,67 @@ export const buildTigerSpecificFunctions = (
...
+                    objectId,
+                });
+                return result.data;
+            });
+        } catch (error: any) {
+            throw convertApiError(error);
+        }
+    },
+
+    setWorkspaceDataFilterSettings: async (
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:1298]: Do we really need to return value for a `setXXX` function? 

There is also no return value for function [setWorkspaceDataFiltersLayout](https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts#L1206C5-L1206C34), is there?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 4]
```diff
@@ -302,6 +306,26 @@ export type DependentEntitiesGraphResponse = DependentEntitiesResponse;
  */
 export type WorkspaceDataFiltersLayout = DeclarativeWorkspaceDataFilters;
 
+/**
+ * @internal
+ */
+export type WorkspaceDataFiltersResult = JsonApiWorkspaceDataFilterOutDocument;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:312]: Please rename `WorkspaceDataFiltersResult` → `WorkspaceDataFilterResult` (without any `s` after `Filter`).

`WorkspaceDataFiltersResult` can make people think that the type is some sort of array, can't it?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 5]
```diff
@@ -302,6 +306,26 @@ export type DependentEntitiesGraphResponse = DependentEntitiesResponse;
...
+
+/**
+ * @internal
+ */
+export type WorkspaceDataFilterSettingsResult = JsonApiWorkspaceDataFilterSettingOutDocument;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:317]: Please rename `WorkspaceDataFilterSettingsResult` → `WorkspaceDataFilterSettingResult` (without any `s` after `Setting`).

`WorkspaceDataFilterSettingsResult` can make people think that the type is some sort of array, can't it?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 6]
```diff
@@ -302,6 +306,26 @@ export type DependentEntitiesGraphResponse = DependentEntitiesResponse;
...
+
+/**
+ * @internal
+ */
+export type WorkspaceDataFilter = JsonApiWorkspaceDataFilterInDocument;
+
+/**
+ * @internal
+ */
+export type WorkspaceDataFilterSettings = JsonApiWorkspaceDataFilterSettingInDocument;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:327]: Please rename `WorkspaceDataFilterSettings` → `WorkspaceDataFilterSetting` (without any `s` after `Setting`).

`WorkspaceDataFilterSettings` can make people think that the type is some sort of array, can't it?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 7]
```diff
@@ -379,6 +403,19 @@ export type TigerSpecificFunctions = {
     ) => Promise<IInvitationUserResponse>;
     getWorkspaceDataFiltersLayout?: () => Promise<WorkspaceDataFiltersLayout>;
     setWorkspaceDataFiltersLayout?: (workspaceDataFiltersLayout: WorkspaceDataFiltersLayout) => Promise<void>;
+    getWorkspaceDataFilters?: (workspaceId: string, objectId: string) => Promise<WorkspaceDataFiltersResult>;
+    setWorkspaceDataFilters?: (
+        workspaceId: string,
+        workspaceDataFilter: WorkspaceDataFilter,
+    ) => Promise<WorkspaceDataFiltersResult>;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:410]: As also mentioned in the other comment, if there is no need to specify the return type for a `setXXX` function. Please change the return type to `Promise<void>`.

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 8]
```diff
@@ -379,6 +403,19 @@ export type TigerSpecificFunctions = {
...
+    getWorkspaceDataFilterSettings?: (
+        workspaceId: string,
+        objectId: string,
+    ) => Promise<WorkspaceDataFilterSettingsResult>;
+    setWorkspaceDataFilterSettings?: (
+        workspaceId: string,
+        workspaceDataFilterSettings: WorkspaceDataFilterSettings,
+    ) => Promise<WorkspaceDataFilterSettingsResult>;
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:418]: As also mentioned in the other comment, if there is no need to specify the return type for a `setXXX` function. Please change the return type to `Promise<void>`.

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 9]
```diff
@@ -379,6 +403,19 @@ export type TigerSpecificFunctions = {
     ) => Promise<IInvitationUserResponse>;
     getWorkspaceDataFiltersLayout?: () => Promise<WorkspaceDataFiltersLayout>;
     setWorkspaceDataFiltersLayout?: (workspaceDataFiltersLayout: WorkspaceDataFiltersLayout) => Promise<void>;
+    getWorkspaceDataFilters?: (workspaceId: string, objectId: string) => Promise<WorkspaceDataFiltersResult>;
+    setWorkspaceDataFilters?: (
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:407]: Please rename `setWorkspaceDataFilters` → `setWorkspaceDataFilter` (without any `s` after `Filter`).

`setWorkspaceDataFilters` can make people think that the function can set an array of workspace data filters, can't it?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 10]
```diff
@@ -379,6 +403,19 @@ export type TigerSpecificFunctions = {
...
+        workspaceId: string,
+        workspaceDataFilter: WorkspaceDataFilter,
+    ) => Promise<WorkspaceDataFiltersResult>;
+    getWorkspaceDataFilterSettings?: (
+        workspaceId: string,
+        objectId: string,
+    ) => Promise<WorkspaceDataFilterSettingsResult>;
+    setWorkspaceDataFilterSettings?: (
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:415]: Please rename `setWorkspaceDataFilterSettings` → `setWorkspaceDataFilterSetting` (without any `s` after `Setting`).

`setWorkspaceDataFilterSettings` can make people think that the function can set an array of workspace data filter settings, can't it?

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 11]
```diff
@@ -379,6 +403,19 @@ export type TigerSpecificFunctions = {
...
+        workspaceId: string,
+        workspaceDataFilterSettings: WorkspaceDataFilterSettings,
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:417]: Please rename `setWorkspaceDataFilterSettings` → `setWorkspaceDataFilterSetting` (without any `s` after `Setting`).

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15718/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17227/ : FAILURE in 6m 51s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15719/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17228/ : FAILURE in 6m 27s

**Comment by anvuong119** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15741/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17240/ : FAILURE in 7m 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15751/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17249/ : SUCCESS in 12m 12s

File: libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts [hunk 12]
```diff
@@ -1210,6 +1244,65 @@ export const buildTigerSpecificFunctions = (
...
+                return result.data;
+            });
+        } catch (error: any) {
+            throw convertApiError(error);
+        }
+    },
+
+    setWorkspaceDataFilterSetting: async (
+        workspaceId: string,
+        workspaceDataFilterSettings: WorkspaceDataFilterSetting,
```
Comment (anvuong119) [inline libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts:1296]: Please rename `workspaceDataFilterSettings` → `workspaceDataFilterSetting` (in singular).

A name in plural might make people think that it is an array.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15754/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17253/ : FAILURE in 6m 30s

**Review by anvuong119 — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4085

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3198/ : SUCCESS in 11m 35s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17255/ : SUCCESS in 13m 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/810/ : SUCCESS in 10m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4357/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1078/ : FAILURE in 1m 22s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/722/ : SUCCESS in 7m 42s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/437/ : FAILURE in 11m 02s

**Comment by thuytran-gooddata** [comment]
ok to test

**Comment by thuytran-gooddata** [comment]
extended test - tiger-cypress - integrated

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by thuytran-gooddata** [comment]
extended test - tiger-cypress - integrated

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15758/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17257/ : SUCCESS in 13m 06s

**Review by anvuong119 — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4085

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3199/ : SUCCESS in 9m 28s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17258/ : SUCCESS in 13m 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/811/ : SUCCESS in 11m 09s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4358/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1079/ : FAILURE in 1m 26s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/723/ : SUCCESS in 8m 39s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/438/ : SUCCESS in 6m 20s

**Pull request merged**